client/state/data-layer/wpcom/sites/posts/likes/new/index.js

```diff
@@ -0,0 +1,51 @@
+/** @format */
+
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+import { unlike, addLiker } from 'state/posts/likes/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { POST_LIKE } from 'state/action-types';
+import { bypassDataLayer } from 'state/data-layer/utils';
+
+export const fetch = action =>
+	http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/posts/${ action.postId }/likes/new`,
+			body: {},
+			apiVersion: '1.1',
+		},
+		action
+	);
+
+export const onSuccess = ( { siteId, postId }, { likeCount, liker } ) =>
+	addLiker( siteId, postId, likeCount, liker );
+
+export const onError = ( { siteId, postId } ) => bypassDataLayer( unlike( siteId, postId ) );
```

Comment:
with `POST_LIKE_REQEUEST` and `POST_UNLIKE_REQUEST` we wouldn't have this condition where we have to bypass the data layer because the data layer is only listening for things that need to sync with the remote